### DISCUSSION TITLE
What's the sharing scope used for?

### DISCUSSION THREAD

**Discussion by michaelsafyan** [gilesknap/gphotos-sync#405] [Q&A]
Looking in `Main.py`, I see that two scopes are requested:

- `https://www.googleapis.com/auth/photoslibrary.readonly`
- `https://www.googleapis.com/auth/photoslibrary.sharing`

What is the `photoslibrary.sharing` scope being used for?

**Comment by gilesknap** [comment]
Hi @michaelsafyan if you turn on family sharing then the shared library will also be downloaded.
EDIT: that was wrong - it is needed to see the contents of shared albums.

**Comment by gilesknap** [comment]
I just reviewed what this scope enables and I can see that there is a security concern there https://developers.google.com/photos/library/guides/authorization. I'm pretty sure I needed this scope to be able to read shared albums but wrote this a long time ago so can't be certain it was required.

gphhotos-sync only does read operations on your library
